keras/engine/training.py

```diff
@@ -3782,6 +3802,9 @@ def reduce_per_replica(values, strategy, reduction="first"):
       ValueError: if the reduction method is not supported.
     """
 
+    if reduction == "auto":
+        reduction = "first" if _is_tpu_multi_host(strategy) else "sum"
```

Comment:
Does it make sense to change `_is_tpu_multi_host()` to `backend.is_tpu_strategy()` because even if there's only one TPU host (worker) there can still be multiple values in a `TPUDistributedVariable`?

Reply:
agreed, done in https://github.com/keras-team/keras/pull/16664/commits/66a30d2a319025ef05bf075eab339d6b7394bba6